src/autocomplete/databases/clickhouse/tests/extract-statement-positions.test.ts

```diff
@@ -0,0 +1,163 @@
+        WHERE id = 1;`;
+    const result = extractClickHouseStatementPositionsFromQuery(query);
+    const expectedResult: StatementPosition[] = [
+        {
+            startIndex: 0,
+            endIndex: 66,
+        },
+    ];
+
+    expect(result).toEqual(expectedResult);
+
+    expect(query.slice(result[0]?.startIndex, result[0]?.endIndex)).toBe(
+        `SELECT * FROM
+            art -- comment here
+        WHERE id = 1`,
+    );
+});
+
+test('should extract statement from query without inlined comment at the end', () => {
+    const query = 'SELECT * FROM art WHERE id = 1; -- comment here';
+    const result = extractClickHouseStatementPositionsFromQuery(query);
+    const expectedResult: StatementPosition[] = [
+        {
+            startIndex: 0,
+            endIndex: 30,
+        },
+    ];
+
+    expect(result).toEqual(expectedResult);
+
+    expect(query.slice(result[0]?.startIndex, result[0]?.endIndex)).toBe(
+        'SELECT * FROM art WHERE id = 1',
+    );
+});
+
+test('should extract unfinished statement', () => {
+    const query = 'SELECT * FROM art WHERE id =';
+    const result = extractClickHouseStatementPositionsFromQuery(query);
+    const expectedResult: StatementPosition[] = [
+        {
+            startIndex: 0,
+            endIndex: 28,
+        },
+    ];
+
+    expect(result).toEqual(expectedResult);
+
+    expect(query.slice(result[0]?.startIndex, result[0]?.endIndex)).toBe(
+        'SELECT * FROM art WHERE id =',
+    );
+});
+
+test('should extract finished and unfinished statements', () => {
+    const query = `SELECT * FROM art WHERE id = 1; SELECT * FROM art WHERE id =`;
+    const result = extractClickHouseStatementPositionsFromQuery(query);
+    const expectedResult: StatementPosition[] = [
+        {
+            startIndex: 0,
+            endIndex: 30,
+        },
+        {
+            startIndex: 32,
+            endIndex: 60,
+        },
+    ];
+
+    expect(result).toEqual(expectedResult);
+
+    expect(query.slice(result[0]?.startIndex, result[0]?.endIndex)).toBe(
+        'SELECT * FROM art WHERE id = 1',
+    );
+    expect(query.slice(result[1]?.startIndex, result[1]?.endIndex)).toBe(
+        'SELECT * FROM art WHERE id =',
+    );
+});
+
+test('should not extract any statement from comments', () => {
+    const query = `
+        -- SELECT * FROM art WHERE id = 1;
+        /* SELECT * FROM art WHERE id = 1; */
+    `;
+    const result = extractClickHouseStatementPositionsFromQuery(query);
+
+    expect(result).toEqual([]);
+});
+
+test('should ignore empty statements', () => {
+    const query = `;;;;;;`;
+    const result = extractClickHouseStatementPositionsFromQuery(query);
+
+    expect(result).toEqual([]);
+});
+
+test('should ignore newlines', () => {
+    const query = '\n\n\n\n';
+    const result = extractClickHouseStatementPositionsFromQuery(query);
+
+    expect(result).toEqual([]);
+});
+
+test('should extract statements from multiple query', () => {
```

Comment:
can we please add one test with 3 statements? this is important code and in theory what works for 2 statements might not work for 3 and more